client_test.go

```diff
@@ -109,6 +114,30 @@ var callTests = []struct {
 	req:         &chM4Req{},
 	expectResp:  new(int),
 	expectError: `GET http://.*/m4: unexpected content type text/plain; want application/json; content: bad response`,
+	assertError: func(c *gc.C, err error) {
+		c.Assert(errgo.Cause(err), gc.FitsTypeOf, (*httprequest.DecodeResponseError)(nil))
+
+		err1 := errgo.Cause(err).(*httprequest.DecodeResponseError)
+		c.Assert(err1.Response, gc.NotNil)
+		data, err := ioutil.ReadAll(err1.Response.Body)
```

Comment:
nope. it's specifically documented that the response body in the error doesn't need to be closed.
(otherwise a naive client would never know to close it)